src/nvim/ex_getln.c

```diff
@@ -2409,6 +2409,32 @@ static void cmdpreview_prepare(CpInfo *cpinfo)
 static void cmdpreview_restore_state(CpInfo *cpinfo)
   FUNC_ATTR_NONNULL_ALL
 {
+  Set(ptr_t) saved_bufs = SET_INIT;
+  Set(ptr_t) saved_wins = SET_INIT;
```

Comment:
Why do you need the sets? The windows and buffers are already guaranteed to be unique.

Reply:
I didn't want to iterate over all the saved buffer and windows each time

Reply:
Oh, nevermind, I misread the code. My bad.